### PR TITLE
Storage June Release

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-common/CHANGELOG.md [hunk 1]
```diff
@@ -1,17 +1,11 @@
 # Release History
 
-## 12.3.0-beta.1 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
+## 12.2.4 (2022-06-07)
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/CHANGELOG.md:3]: If no package that depends on common is GA'ing yet, is it necessary to GA this right now? Given blobs is shipping beta, maybe this should be as well.

Is the bug fix critical for customers on older GA releases of the SDK?

Comment (ahsonkhan) [reply sdk/storage/azure-storage-common/CHANGELOG.md:3]: That's true, but it seems strange to rev and ship a new version on a shared package which no end customer using GA'd storage service packages can benefit from. That said, it's good that we are only, and correctly, only incrementing the patch version.

Do we even have to ship storage common at all this month, or can we skip it?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/CHANGELOG.md:3]: The new blob release 12.5.0 beta1 depends on it. So we have to ship a common package, either GA or beta.

**Review by RickWinter — APPROVED** [review]

File: sdk/storage/azure-storage-common/CHANGELOG.md (hunk 1 above)
Comment (ahsonkhan) [reply sdk/storage/azure-storage-common/CHANGELOG.md:3]: Alright, it's your call :)

**Review by ahsonkhan — APPROVED** [review]

**Pull request merged**